tests/integration/conftest.py

```diff
@@ -317,3 +326,30 @@ def _get_multi_source_marks(multi_source_test_config: MultiSourceTestConfig) ->
         raise ValueError(
             "MultiSourceBatch tests must either use the same backend or include sqlite."
         )
+
+
+@pytest.fixture(scope="session")
+def test_session_sql_engine_manager():
+    logger.info("SessionSqlEngineManager: Starting setup.")
+    manager = SessionSQLEngineManager()
+    yield manager
+
+    logger.info("SessionSqlEngineManager: Starting teardown.")
+    pre_cleanup_stats = manager.get_all_pool_statistics()
+    # We temporarily log a warning so we can see this in the pytest output without turning on info
+    # logging across the whole test run.
+    logger.warning(
```

Comment:
I log a warning here to see pool statistics.

Reply:
I get that you want to see this, but it's a little confusing. Also, why doesn't this raise an error since we aren't ignoring it in pyproject.toml? Could this be achieved via stdout?

Reply:
I guess you don't need to ignore logs, just `warnings.warn()`.

Reply:
I agree this isn't ideal. I had AI try to write a custom log handler (and in a different attempt write to stderr). It ended up saying this code is never called, but I see it being executed with the pyproject.toml updates. I could try more but I know if it's worth it so left this comment for now.

Reply:
I played a little more and got ai to generate a way to write to stdout in an isolated test case. I may try to simplify it and incorporate it, but if I do, it will be in a separate PR.